--- ../kippo/commands/wget.py
+++ ../kippo/commands/wget.py
@@ -116,8 +116,8 @@
             error = error.getErrorMessage()
         self.writeln(error)
         # Real wget also adds this:
-        #self.writeln('%s ERROR 404: Not Found.' % \
-        #    time.strftime('%Y-%m-%d %T'))
+        self.writeln('%s ERROR 404: NOT FOUND.\n' % \
+            time.strftime('%Y-%m-%d %T'))
         self.exit()
 commands['/usr/bin/wget'] = command_wget
